### PR TITLE
feat: config traversal

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: query/config_traversal.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		}
+
+		typeIDs, err := ConfigIDsByTypeFromCache(ctx, relType)
+		if err != nil {
+			return configItem, fmt.Errorf("no type %s exists for any config: %w", relType, err)
+		}
+
+		configID, ok := lo.Find(relatedIDs, func(relID string) bool {
+			return lo.Contains(typeIDs, relID)
+		})
```
Comment (moshloop) [inline query/config_traversal.go:34]: I think this is going to be really slow, can we not cache it directly e.g. 

`{id}/Kubernetes::Node []string`

File: query/config_traversal.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/flanksource/duty/context"
+	"github.com/flanksource/duty/models"
+	"github.com/samber/lo"
+)
+
+func TraverseConfig(ctx context.Context, id, relationType string) (models.ConfigItem, error) {
```
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply query/config_traversal.go:11]: Thought we can keep an error to highlight if part of the traversal is correct and other part wrong, ie. if you do `Deployment/HelmRelease/Kustomize` and `Deplyoment/HelmRelease` is correct but  HelmRelease has no Kustomize then the error message should reflect that

And we can handle what to return in templating in the wrapper func

What do you think ?

File: query/config_traversal.go (hunk 1 above)
Comment (yashmehrotra) [reply query/config_traversal.go]: Done 👍🏽

File: query/config_traversal.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply query/config_traversal.go]: Okay

File: query/config_cache.go [hunk 3]
```diff
@@ -156,3 +159,84 @@ func FindConfigForComponent(ctx context.Context, componentID, configType string)
...
+	return "configID:" + id
+}
+
+// <config_id> -> []related_ids
+var configRelationCache = cache.New[[]string](gocache_store.NewGoCache(gocache.New(6*time.Hour, 6*time.Hour)))
+
+func configRelationCacheKey(id string) string {
+	return "configRelatedIDs:" + id
+}
+
```
Comment (yashmehrotra) [inline query/config_cache.go:183]: 10 mins for cache time, job runs every 5 mins

Comment (moshloop) [reply query/config_cache.go:183]: @yashmehrotra is there a commit missing ?

Comment (yashmehrotra) [reply query/config_cache.go:183]: Sorry, forgot to push the commit

**Pull request merged**